src/test/java/JsonParserTests.java

```diff
@@ -0,0 +1,93 @@
+import org.junit.jupiter.api.*;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.MethodSource;
+import parser.JsonParser;
+import parser.NoSuchFileException;
+import shop.Cart;
+import shop.RealItem;
+import shop.VirtualItem;
+
+import java.io.File;
+import java.util.stream.Stream;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+@DisplayName("Tests for JsonParser class")
+public class JsonParserTests {
+
+    private JsonParser jsonParser;
+    private Cart cart;
+    private final String FILE_PATH = "src/main/resources/TestCart.json";
+
+    @BeforeEach
+    public void setup() {
+        jsonParser = new JsonParser();
+        cart = new Cart("TestCart");
+        RealItem realItem = new RealItem();
+        realItem.setName("RealItem1");
+        realItem.setPrice(10.0);
+        realItem.setWeight(2.5);
+
+        VirtualItem virtualItem = new VirtualItem();
+        virtualItem.setName("VirtualItem1");
+        virtualItem.setPrice(20.0);
+        virtualItem.setSizeOnDisk(100.0);
+
+        cart.addRealItem(realItem);
+        cart.addVirtualItem(virtualItem);
+        cart.showItems();
+    }
+
+    @AfterEach
+    public void cleanup() {
+        new File(FILE_PATH).delete();
+    }
+
+    @Test
+    @Tag("Json Parser")
+    public void testWriteToFile() {
+        File file = new File(FILE_PATH);
+
+        // Write to file
+        jsonParser.writeToFile(cart);
+
+        assertTrue(file.exists(), "The file should be created");
```

Comment:
as an example, you can also check that `writeToFile()` method writes file correctly. Ex.:
```
private static final String EXPECTED_TEST_CART = "{"cartName":"TestCart","realItems":[{"weight":2.5,"name":"RealItem1","price":10.0}],"virtualItems":[{"sizeOnDisk":100.0,"name":"VirtualItem1","price":20.0}],"total":36.0}"

// some code of the test
String actualCart = Files.readString(Path.of(FILE_PATH));
assertEquals(EXPECTED_TEST_CART, actualCart, "some assertions message");
```